models/tui.go

```diff
@@ -33,30 +35,176 @@ func (m taskGroupsModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	switch msg := msg.(type) {
 	case tea.WindowSizeMsg:
 		h, v := styles.DocStyle.GetFrameSize()
-		m.list.SetSize(msg.Width-h, msg.Height-v)
+		m.list.SetSize(msg.Width-h, msg.Height-v-4)
 	case tea.KeyMsg:
 		switch keyPress := msg.String(); keyPress {
 		case "ctrl+c":
 			return m, tea.Quit
 		}
 		switch m.state {
 		case 0:
-			m, cmd, check := taskGroupViewState0(m, msg)
-			if !check {
-				return m, cmd
```

Comment:
What was the rational behind reversing the split up again? Was it too convoluted?
Can you explain the thought process?